x-pack/legacy/plugins/reporting/server/lib/create_worker.ts

```diff
@@ -31,34 +31,44 @@ export function createWorkerFactory(server: ServerFacade) {
   const { exportTypesRegistry } = server.plugins.reporting!;
 
   // Once more document types are added, this will need to be passed in
-  return function createWorker(queue: ESQueueInstance) {
+  return function createWorker(queue: ESQueueInstance<JobParamsType, JobDocPayloadType>) {
     // export type / execute job map
-    const jobExecutors: Map<string, ESQueueWorkerExecuteFn | ImmediateExecuteFn> = new Map();
+    const jobExecutors: Map<string, any> = new Map();
```

Comment:
Is there a way to avoid using `any` here?
Are the executor always functions? We can at least check that, even if we don't know the signature using a generic `Function` type

Reply:
40484082b8